tests/lib/newlib/newlib_hook/src/main.c

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright (c) 2021 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <ztest.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <math.h>
+#include <sys/unistd.h>
+#include <sys/signal.h>
+#include <sys/stat.h>
+
+/**
+ * @brief Verify _open() stub in newlib libc-hooks.c
+ *
+ * @details The compiler warning when call open(),
+ * use fopen() instead.
+ */
+void test_newlib_stub_open(void)
+{
+	FILE *fp = NULL;
+
+	fp = fopen("../prj.conf", "r");
+	zassert_is_null(fp, "open file error");
+}
+
+/**
+ * @brief Verify _close() stub in newlib libc-hooks.c
+ *
+ * @details The default weak function return -1.
+ */
+void test_newlib_stub_close(void)
+{
+	int ret = 0;
+
+	ret = close(-1);
+	zassert_equal(ret, -1, "closed a invalid fd");
+}
+
+/**
+ * @brief Verify _getpid() stub in newlib libc-hooks.c
+ *
+ * @details The default weak function return 0.
+ */
+void test_newlib_stub_getpid(void)
+{
+	pid_t pid;
+
+	pid = getpid();
+	zassert_false(pid, "getpid failed");
+}
+
+/**
+ * @brief Verify _stat() stub in newlib libc-hooks.c
+ *
+ * @details The default weak function return 0.
+ */
+void test_newlib_stub_fstat(void)
+{
+	struct stat status;
+	int ret;
+
+	ret = fstat(-1, &status);
```

Comment:
Include `fcntl.h` and test `stat()` directly.

Reply:
I included <sys/stat.h> and tried to test stat() directly, but the compiler print
"lib/32/soft-float/libc.a(lib_a-statr.o): in function `_stat_r':
statr.c:(.text._stat_r+0x19): undefined reference to `_stat'
" 
In libc-hook.c, the stub name is "int _fstat(int file, struct stat *st)". 
Is there more appropriate to test fstat() here?